.gitignore

```diff
@@ -1,7 +1,14 @@
-.idea
+.idea/
+
+package-lock.json
```

Comment:
Never ignore lock files, it's a bad practice.

Reply:
If that is the only argument you can come up with you should stop reading so many other people's opinion blog posts. I usually don't check lock files in to see if I can trust package maintainers.

If you want the file to be part of the repository, then just say so, but don't tell me it is bad practice without explaining why...

Reply:
@firstred 
Example:

Form yarn:
https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/
> Without lockfiles it gets even more complicated: In applications or libraries, if there is no lockfile, you will have to check the dependencies every time you install or re-install them and make sure that everything still works. Otherwise the build might be broken or the tests might fail. You could break something without even realizing it. You could run into situations where code works on your machine but no one else’s.

Composer for PHP:
https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control

Gemfile for Ruby:
https://bundler.io/v1.17/guides/faq.html#committing-lockfiles-in-libraries

From stackoverflow:
https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files/44904494
https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

Reply:
Thanks for making my point!

I have closed the pull request because you seem to be more focused on crappy lock files rather than the actual PR itself. I have even given you write permissions to the branch, making you able to polish the tiny bits until it's perfect. These kinds of discussions, and the response time in particular, really give me a good impression of where your priorities lie.

Reply:
@firstred 

- About the response time: the team is busy. The last 3 weeks, 2 security versions and one beta have been released. Plus, the 1th of November is off in France. And some devs also attended the PHP Forum, 2 days of conferences. The devs and the QA team are reviewing PR during their work hours. They are usually constructive and polite with their comments.
- About best practices: currently, the PrestaShop open source project is using best practices as much as possible. This is a prerequisite in order to invest in good quality code, easy to test and to maintain in the future.

I can understand some frustration, however, here, two devs took time to explain, gave details and links about what is currently expected and why.

The aim is really to be able to merge your PR. Otherwise nobody would actually take time to answer to you.

Reply:
@ttoine  ,

This is the last time my businesspartner @firstred ever offers you thousands of € worth of work, which solves all your problems. Handed to you on a golden platter, and this your gratitude? Your team is busy.... All problems solved, all you have to do is a few tweaks left and right and tadaa: done.

This shows the incompetence of your company and the people working there. All pull requests have been removed. If you are so great, go solve / save this platform yourself.

Reply:
Then, tell your "business partner" that when he wants to contribute to any open source project, the aim is not just to solve a problem, but also to write code that will be easy to maintain in the future by other developers. That is why following best practices is mandatory. That is what we tried to explain here.

And honestly, I can't believe that the team get this kind of answers, although they are asking for quality and details. Again, the aim is to merge the PRs of the community, and many other contributors see their PR merged: they follow the docs, the best practices, the code style, like the PrestaShops devs team (they follow the same rules without exception). Establishing a good relationship and collaborating on GitHub is actually easy and based on good will.

That said, the doors of the PrestaShop open source project will always stay open to respectful and polite contributors.

Reply:
What a strange reactions to so many unique chances to improve the software.

Do as you please; you have my vote of no confidence. I will not contribute anymore to this project.